### ISSUE TITLE
Ability to ignore books based on editions

### ISSUE THREAD

**Issue by plmcgrn** [Readarr/Readarr#1508]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

In a lot of scenarios, I'm finding that books are getting added to authors where the only edition available at present has Kindle one, and I'm only intending to use Readarr for audiobooks.  I'd either like to blanket ignore any editions that aren't audio, or at least determine Missing status based on the release date of those editions, not the printed edition release dates.

Example as of 1/31/21
https://www.goodreads.com/work/editions/92844369

I'd rather Readarr ignore this book entirely, until an audiobook edition is released.

### Describe the solution you'd like

At present, the Release Date is used to drive whether a book is considered Missing or not.  This is fine for normal print editions, but audiobooks tend to trail the print releases for a lot of authors.  This feature request is to look at the release date of Editions, instead of "whatever release data the default Goodreads Edition of a book has".  Alternatively, it could just ignore books that don't/won't match the quality profile I have set (in this case, "Spoken").

### Describe alternatives you've considered

Not aware of any way to ignore books based on metadata, because the search source is using Goodreads data, but the relevant settings really only apply to downloading.  The search (both automatic via watch settings and manual) have no way to filter out books based on Goodreads' book editions.

### Anything else?

n/a

[AB#2524](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2524)

**Comment by bakerboy448** [comment]
This should be tied into only matching audiobooks and ebooks based on the applicable quality / quality profile

**Comment by plmcgrn** [comment]
> This should be tied into only matching audiobooks and ebooks based on the applicable quality / quality profile

That too, yeah!  I prefer M4B to MP3, so I've only been doing manual searches for files, to ensure I don't end up with a collection of MP3's when M4B's are available for the same book on the same indexer.

**Comment by bakerboy448** [comment]
> That too, yeah! I prefer M4B to MP3, so I've only been doing manual searches for files, to ensure I don't end up with a collection of MP3's when M4B's are available for the same book on the same indexer.

you just need your quality profile configured appropriately; this is automatically done. have m4b above mp3

**Comment by plmcgrn** [comment]
> you just need your quality profile configured appropriately; this is automatically done. have m4b above mp3

Thanks for that, I've modified it.  I haven't set Readarr to auto-download, yet, but once I finish organizing my massive existing library within it, I probably will do that.  Would M4B be considered an "upgrade" to MP3, if I prioritize it accordingly?

**Comment by bakerboy448** [comment]
Yes - https://wiki.servarr.com/readarr/settings#quality-profiles

**Comment by ashanrath** [comment]
> Yes - https://wiki.servarr.com/readarr/settings#quality-profiles

That doesn't prevent the book from being added, only downloaded. What would be ideal is a metadata filter that would only add booked based on certain criteria. E.g. If you are using readarr for ebooks, exclude editions with "audio" or something similar. If you are using readarr for audiobooks, only include editions with "audio", "CD" etc.

**Comment by plmcgrn** [comment]
> > Yes - https://wiki.servarr.com/readarr/settings#quality-profiles
> 
> That doesn't prevent the book from being added, only downloaded. What would be ideal is a metadata filter that would only add booked based on certain criteria. E.g. If you are using readarr for ebooks, exclude editions with "audio" or something similar. If you are using readarr for audiobooks, only include editions with "audio", "CD" etc.

This is already possible.  I'm using Readarr for audiobooks and removed the eBook formats from the profile.  This Issue is more about ignoring editions so that "Kindle" editions don't get added as Monitored books, since many times the print/digital version of a book will be released months/years before an audiobook version.  In that scenario, the book shows as "missing" in Readarr, but in reality, that edition isn't available yet.

**Comment by InterferencePattern** [comment]
I don't know if the Goodreads API allows for filtering by edition type (audiobook, etc.), but it would be nice to have a way to only show books under "Wanted" if they exist in the first place.
